Include/internal/pycore_code.h

```diff
@@ -24,6 +26,103 @@ struct _PyOpcache {
     char optimized;
 };
 
+
+/* PEP 659
+ * Specialization and quickening structs and helper functions
+ */
+
+typedef struct {
+    int32_t cache_count;
+    int32_t _; /* Force 8 byte size */
+} _PyEntryZero;
+
+/* Add specialized versions of entries to this union.
+ *
+ * Do not break the invariant: sizeof(SpecializedCacheEntry) == 8
```

Comment:
Maybe this comment should be next to the assertion on this invariant rather than here.